izanami-frontend/src/components/SearchComponant/SearchResults.tsx

```diff
@@ -0,0 +1,191 @@
+    );
+    if (allHistory[username]) {
+      setSearchHistory(allHistory[username]);
+    }
+  }, []);
+
+  const handleItemClick = (item: SearchResult) => {
+    const tenantExists = item.path.some((element) => element.type === "tenant");
+
+    if (!tenantExists) {
+      item.path = [...item.path, { type: "tenant", name: item.tenant }];
+    }
+
+    setSearchHistory((prevHistory) => {
+      const allHistory = JSON.parse(
+        localStorage.getItem("userSearchHistory") || "{}"
+      );
+      const userHistory = allHistory[username] || [];
+      const updatedHistory = Array.from(
+        new Map(
+          [...userHistory, ...prevHistory, item].map((item) => [
+            item.name + item.type + item.tenant, // better to have item id
+            item,
+          ])
+        ).values()
+      ).slice(-30);
+      allHistory[username] = updatedHistory;
+      localStorage.setItem("userSearchHistory", JSON.stringify(allHistory));
+
+      return updatedHistory;
+    });
+    onClose();
+  };
+  const filteredHistory = searchHistory
+    .filter(
+      (item) =>
+        (modalStatus.all || item.tenant === modalStatus.tenant) &&
+        (filterOptions.length === 0 || filterOptions.includes(item.type))
+    )
+    .slice(-5);
+
+  return (
+    <div className="search-container">
+      {filteredHistory.length > 0 && resultStatus.state == "INITIAL" && (
+        <div className="search-result">
+          <>
+            <span>Recent</span>
+            <ul className="search-ul nav flex-column">
+              {filteredHistory.map((term, index) => (
+                <li className="search-ul-item" key={index}>
+                  <Link
+                    to={getLinkPath(term) || "#"}
+                    onClick={() => handleItemClick(term)}
+                  >
+                    <ul
+                      className="breadcrumb"
+                      style={{
+                        marginBottom: "0rem",
+                      }}
+                    >
+                      {modalStatus.all &&
+                        term.path.map((pathElement, pathIndex) => (
+                          <li
+                            className="breadcrumb-item"
+                            key={`${term.name}-${pathIndex}`}
+                          >
+                            {typeDisplayInformation
+                              .get(pathElement.type)
+                              ?.icon() ?? ""}
+                            {pathElement.name}
+                          </li>
+                        ))}
+                      <li className="breadcrumb-item" key={`${term.name}-name`}>
+                        {typeDisplayInformation.get(term.type)?.icon() ?? ""}
+                        {term.name}
+                      </li>
+                    </ul>
+                  </Link>
+                </li>
+              ))}
+            </ul>
+          </>
+        </div>
+      )}
+
+      {resultStatus.state !== "INITIAL" && (
```

Comment:
This whole ternary expression imbrication is quite hard to read and understand, I think an if / else if / else statement would be best here.